File: in_Java/0108 Convert Sorted Array to Binary Search Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        if(nums == null || nums.length == 0) return null;
        int i = 0, j = nums.length-1;
        int m = i + (j-i)/2;
        TreeNode root = new TreeNode(nums[m]);   
        int[] left = new int[m];
        int[] right = new int[j-m];
        for(int id = 0; id <= m-1; id++) left[id] = nums[id];
        for(int idd = 0; idd < j-m; idd++) right[idd] = nums[idd + m + 1];        
        root.left = sortedArrayToBST(left);
        root.right = sortedArrayToBST(right);
        return root;
    }
}

File: in_Python/0108 Convert Sorted Array to Binary Search Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def sortedArrayToBST(self, nums):
        """
        :type nums: List[int]
        :rtype: TreeNode
        """
        if nums == None or len(nums) == 0:
            return None
        i,j = 0, len(nums)-1
        if i == j:
            return TreeNode(nums[i])
        m = i + (j-i)/2
        root = TreeNode(nums[m])
        root.left = self.sortedArrayToBST(nums[0:m])
        root.right = self.sortedArrayToBST(nums[(m+1) : len(nums)])
        return root